kubernetes/test-infra
# kubetest: add EKS plugin

https://github.com/aws/awstester implements https://github.com/kubernetes/test-infra/blob/9a88e32c98e4897cdb7ada607d440d2323ad221a/kubetest/main.go#L223-L230

Example Prow configuration for ALB Ingress Controller:

```yaml
presets:
- env:
  - name: AWSTESTER_EKS_KUBETEST_EMBEDDED_BINARY
    value: "true"
  - name: AWSTESTER_EKS_WAIT_BEFORE_DOWN
    value: 1m0s
  - name: AWSTESTER_EKS_DOWN
    value: "true"
  - name: AWSTESTER_EKS_ENABLE_NODE_SSH
    value: "true"
  - name: AWSTESTER_EKS_AWSTESTER_IMAGE
    value: 607362164682.dkr.ecr.us-west-2.amazonaws.com/awstester:180919694a39
  - name: AWSTESTER_EKS_WORKER_NODE_INSTANCE_TYPE
    value: m3.xlarge
  - name: AWSTESTER_EKS_WORKER_NODE_ASG_MIN
    value: "1"
  - name: AWSTESTER_EKS_WORKER_NODE_ASG_MAX
    value: "1"
  - name: AWSTESTER_EKS_ALB_ENABLE
    value: "true"
  - name: AWSTESTER_EKS_ALB_ALB_INGRESS_CONTROLLER_IMAGE
    value: quay.io/coreos/alb-ingress-controller:1.0-beta.7
  - name: AWSTESTER_EKS_ALB_TARGET_TYPE
    value: instance
  - name: AWSTESTER_EKS_ALB_TEST_MODE
    value: nginx
  - name: AWSTESTER_EKS_ALB_TEST_SCALABILITY
    value: "true"
  - name: AWSTESTER_EKS_ALB_TEST_SERVER_REPLICAS
    value: "1"
  - name: AWSTESTER_EKS_ALB_TEST_SERVER_ROUTES
    value: "1"
  - name: AWSTESTER_EKS_ALB_TEST_CLIENTS
    value: "20"
  - name: AWSTESTER_EKS_ALB_TEST_CLIENT_REQUESTS
    value: "2000"
  - name: AWSTESTER_EKS_ALB_TEST_RESPONSE_SIZE
    value: "40960"
  - name: AWSTESTER_EKS_ALB_TEST_CLIENT_ERROR_THRESHOLD
    value: "500"
  - name: AWSTESTER_EKS_ALB_TEST_EXPECT_QPS
    value: "100.000000"
  - name: AWS_SHARED_CREDENTIALS_FILE
    value: /etc/aws-cred-awstester/aws-cred-awstester
  labels:
    preset-e2e-aws-alb-ingress-controller: "true"
  volumeMounts:
  - mountPath: /etc/aws-cred-awstester
    name: aws-cred-awstester
    readOnly: true
  volumes:
  - name: aws-cred-awstester
    secret:
      secretName: aws-cred-awstester

presubmits:
  gyuho/aws-alb-ingress-controller:
  - agent: kubernetes
    always_run: true
    context: test-presubmit
    labels:
      preset-dind-enabled: "true"
      preset-e2e-aws-alb-ingress-controller: "true"
    name: pull-ingress-aws-alb-build
    rerun_command: /test pull-ingress-aws-alb-build
    run_after_success:
    - agent: kubernetes
      always_run: true
      context: test-presubmit
      labels:
        preset-dind-enabled: "true"
        preset-e2e-aws-alb-ingress-controller: "true"
      name: pull-ingress-aws-alb-e2e
      rerun_command: /test pull-ingress-aws-alb-e2e
      spec:
        containers:
        - command:
          - tests/alb-e2e.sh
          image: 607362164682.dkr.ecr.us-west-2.amazonaws.com/awstester:db7d4d8ed97c
          name: ""
          resources: {}
          securityContext:
            privileged: true
      trigger: (?m)^/test pull-ingress-aws-alb-e2e
    spec:
      containers:
      - command:
        - /bin/echo
        - pull-ingress-aws-alb-build
        image: alpine
        name: ""
        resources: {}
        securityContext:
          privileged: true
    trigger: (?m)^/test pull-ingress-aws-alb-build
```

The configuration above has been working well with our internal Prow cluster (see https://github.com/gyuho/aws-alb-ingress-controller/pull/3 for example output).

Now, we would like to integrate with upstream.

Action items (with following PRs):
- Change `DumpClusterLogs` and `Publish` methods to use upstream Google Cloud Storage?
- Mount AWS test account credential to upstream Prow cluster
- Figure out how to use "dind" for building [aws-alb-ingress-controller](https://github.com/kubernetes-sigs/aws-alb-ingress-controller) images
  - Just like [containerd-cri-presubmit-jobs.yaml](https://github.com/kubernetes/test-infra/blob/master/config/jobs/containerd/cri/containerd-cri-presubmit-jobs.yaml)
- Add Prow job for [aws-alb-ingress-controller](https://github.com/kubernetes-sigs/aws-alb-ingress-controller)
- Add [aws-alb-ingress-controller](https://github.com/kubernetes-sigs/aws-alb-ingress-controller) tests to upstream [test grid](https://github.com/kubernetes/test-infra/tree/master/testgrid)
- Add Prow job for [aws-ebs-csi-driver](https://github.com/kubernetes-sigs/aws-ebs-csi-driver)
- Add [aws-ebs-csi-driver](https://github.com/kubernetes-sigs/aws-ebs-csi-driver) tests to upstream [test grid](https://github.com/kubernetes/test-infra/tree/master/testgrid)
- Add Prow job for [cloud-provider-aws](https://github.com/kubernetes/cloud-provider-aws)
- Add [cloud-provider-aws](https://github.com/kubernetes/cloud-provider-aws) tests to upstream [test grid](https://github.com/kubernetes/test-infra/tree/master/testgrid)
- Add `pull-kubernetes-e2e-aws-eks` Prow job
